77 firebase sync for database

semik997 opened the pull request.

semik997 committed:
Added pod file, Fixed warning

semik997 committed:
Fixed bug with long loading after request

semik997 committed:
test commit

semik997 committed:
Added SDWebImage library and
did image uploading with it

semik997 committed:
Connect saving images in Firebase

semik997 committed:
Added safe extraction force unwrap

semik997 committed:
Refactor code

semik997 committed:
Redesigned settings screen
Corrected the screen with detailed information
Refactor code

semik997 committed:
Refactor code

semik997 committed:
Refactor code

semik997 committed:
Removed coloring of the like button in Favorite View

semik997 committed:
Added comments to the code
Made logical indents
Reduced the number of returns
Moved all numbers to constants
Optimized the code

semik997 committed:
Removed extra podfile

semik997 committed:
Added the missing parts of the code after the rebase

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -27,7 +27,11 @@ class FavoritesCollectionVC: UICollectionViewController, UIPopoverPresentationCo
     private var chooseSize: SettingViewController.ChooseSize?
     private var defaultSizeCell: CGSize?
-    var context: NSManagedObjectContext?
+    private let insret = UIEdgeInsets(top: 50, left: 0, bottom: 100, right: 0)
```
Опечатка.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -46,11 +50,12 @@ class FavoritesCollectionVC: UICollectionViewController, UIPopoverPresentationCo
     }
     
-    // MARK: - UICollectionViewDataSource
+    // MARK: - Setting the number of cells and their filling
```
Лучше верни предыдущую марку, она сама себя поясняет и так.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -129,19 +140,19 @@ class FavoritesCollectionVC: UICollectionViewController, UIPopoverPresentationCo
         }
-        alertController.addAction(yes)
-        alertController.addAction(no)
+        alertController.addAction(yesButton)
```
Можешь попробовать не создавать кнопки для алерта отдельно, а создавать их прямо в addAction. Это будет немного чище (ты сам это осознаешь если сделаешь).

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -196,30 +207,30 @@ extension FavoritesCollectionVC: FavoriteDeleteProtocol {
         switch self.chooseSize {
         case .big:
-            let sizeCell = CGSize (width: 400, height: 400)
+            let sizeCell = bigSize
```
А какой смысл создавать здесь переменную с размером, если ты можешь просто сразу засетить defaultSizeCell используя одну из переменных для размера?

semik997 replied:
Переменная, что бы вынести размер в константу
Не совсем понимаю " если ты можешь просто сразу засетить defaultSizeCell используя одну из переменных для размера?"

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -22,15 +24,18 @@ class MainCollectionVC: UICollectionViewController, UIPopoverPresentationControl
     private let defaults = UserDefaults.standard
     private var tap = UITapGestureRecognizer()
     private var searchText = ""
+    private let insret = UIEdgeInsets(top: 50, left: 0, bottom: 100, right: 0)
```
Опечатка.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -194,25 +222,25 @@ extension MainCollectionVC: UICollectionViewDelegateFlowLayout {
         switch self.chooseSize {
         case .big:
-            let sizeCell = CGSize (width: 400, height: 400)
+            let sizeCell = bigSize
```
Аналогично с предыдущим комментарием про размеры.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -183,6 +198,19 @@ extension MainCollectionVC: FavoriteDeleteProtocol {
+                guard let imageData = image.jpegData(compressionQuality: 1.0) else { return }
+                let metadata = StorageMetadata()
+                metadata.contentType = "image/jpeg"
+                reference.putData(imageData, metadata: metadata)
```
А putData выполняется асинхронно или нет?

semik997 replied:
Да, асинхронно

PaszhaBiceps commented on Movie Search App/Model+Constants/Constants.swift:
```diff
@@ -23,5 +24,12 @@ let placeholderFilm = "https://www.salonlfc.com/wp-content/uploads/2018/01/image
+//MARK: - Other constants
+let interval = 0.1
+let maxValue: Float = 1
+let addValue: Float = 0.2
```
Поработай на неймингом этих констант. Сейчас не понятно для чего они используются.

PaszhaBiceps commented on Movie Search App/SplashScreen+Reachability/SplashScreen.swift:
```diff
@@ -18,9 +18,9 @@ class LoadViewController: UIViewController {
-        Timer.scheduledTimer(withTimeInterval: 0.1, repeats: true) { timer in
-            if self.progress.progress != 1 {
-                self.progress.progress += 0.05
+        Timer.scheduledTimer(withTimeInterval: interval, repeats: true) { timer in
```
Если константы interval, maxValue и addValue используются только на SpalshViewController - вынеси их в него как приватные проперти. Нет смысла их держать в общем доступе в таком случае.

PaszhaBiceps commented on Movie Search App/TableVC/SettingViewController.swift:
```diff
@@ -32,34 +32,57 @@ class SettingViewController: UIViewController, UIColorPickerViewControllerDelega
+        smallSizeCellButton.isSelected = false
         self.delegate?.updateInterface(color: color, size: size)
     }
     @IBAction private func chooseSmallSizeButton(_ sender: UIButton) {
```
Касательно этих экшенов, ты можешь подкинуть все кнопки к одному IBAction и избежать тем самым дублирования.

PaszhaBiceps reviewed (changes requested):
Оставил несколько комментов. И со следующего пул реквеста перейдем на английский 😏

semik997 committed:
Connected all size selection buttons to one IBAction for code optimization
Changed the declaration of AlertController
Changed the names of the constants and moved them to the class where they are used
Fixed typos

semik997 committed:
Made a choice of size through the switch
Removed an extra variable in the size of the collection

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -30,7 +30,7 @@ class MainCollectionVC: UICollectionViewController, UIPopoverPresentationControl
     private var tap = UITapGestureRecognizer()
     private var searchText = ""
-    private let insret = UIEdgeInsets(top: 50, left: 0, bottom: 100, right: 0)
+    private let inseret = UIEdgeInsets(top: 50, left: 0, bottom: 100, right: 0)
```
Опять опечатка.

semik997 replied:
Fixed

semik997 committed:
Fixed typos

semik997 committed:
Added the ability to hide the keyboard when scrolling the screen

semik997 committed:
Fixed saving images in Firebase

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -30,7 +30,7 @@ class MainCollectionVC: UICollectionViewController, UIPopoverPresentationControl
     private var tap = UITapGestureRecognizer()
     private var searchText = ""
-    private let inseret = UIEdgeInsets(top: 50, left: 0, bottom: 100, right: 0)
+    private let insert = UIEdgeInsets(top: 50, left: 0, bottom: 100, right: 0)
```
Опечатка.

semik997 committed:
Fixed typos

PaszhaBiceps reviewed (approved):
Опять опечатка, но я уже устал.
